components/ShoppingItemForm/ShoppingItemForm.js

```diff
@@ -0,0 +1,134 @@
+import { useState } from "react";
+import categories from "@/utils/categories.json";
+import {
+  StyledFormContainer,
+  StyledFormTitle,
+  StyledInputField,
+  StyledLabel,
+  StyledInput,
+  StyledSelect,
+  StyledTextarea,
+  StyledButton,
+  StyledErrorMessage,
+} from "./StyledShoppingItemForm";
+
+const errorMessages = {
+  name: "Name is required",
+  quantity: "Quantity must be a number greater than 1",
+  category: "Please select a category",
+};
+
+export default function ShoppingItemForm({ onAddItem, setShowForm }) {
+  const [name, setName] = useState("");
+  const [quantity, setQuantity] = useState("1");
+  const [category, setCategory] = useState("");
+  const [comment, setComment] = useState("");
+  const [errors, setErrors] = useState({});
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    let formErrors = {};
+
+    if (!name) formErrors.name = errorMessages.name;
+    if (!quantity || isNaN(quantity) || quantity <= 0)
+      formErrors.quantity = errorMessages.quantity;
+    if (!category) formErrors.category = errorMessages.category;
+
+    if (Object.keys(formErrors).length > 0) {
+      setErrors(formErrors);
+      return;
+    }
+
+    const newItem = {
+      id: Date.now().toString(),
```

Comment:
After implementing the suggestions for the `_app.js`, you don't need to set the id here anymore:

```suggestion
```